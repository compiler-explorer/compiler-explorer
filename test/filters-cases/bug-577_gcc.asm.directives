.Ltext0:
_Z6squarei:
.LFB0:
        push    rbp
        mov     rbp, rsp
        mov     DWORD PTR -4[rbp], edi
#APP
# 3 "example.cpp" 1
label:
# 0 "" 2
#NO_APP
        mov     eax, DWORD PTR -4[rbp]
        imul    eax, DWORD PTR -4[rbp]
        pop     rbp
        ret
.LFE0:
.Letext0:
.Ldebug_info0:
.Ldebug_abbrev0:
.Ldebug_line0:
.LASF1:
.LASF3:
.LASF2:
.LASF0:
