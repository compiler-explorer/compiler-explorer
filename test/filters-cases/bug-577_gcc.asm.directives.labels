_Z6squarei:
        push    rbp
        mov     rbp, rsp
        mov     DWORD PTR -4[rbp], edi
#APP
# 3 "example.cpp" 1
# 0 "" 2
#NO_APP
        mov     eax, DWORD PTR -4[rbp]
        imul    eax, DWORD PTR -4[rbp]
        pop     rbp
        ret