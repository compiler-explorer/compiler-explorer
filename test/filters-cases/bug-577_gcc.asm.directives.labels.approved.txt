{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "_Z6squarei:"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        push    rbp"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        mov     rbp, rsp"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        mov     DWORD PTR -4[rbp], edi"
    },
    {
      "labels": [],
      "source": null,
      "text": "#APP"
    },
    {
      "labels": [],
      "source": null,
      "text": "# 3 \"example.cpp\" 1"
    },
    {
      "labels": [],
      "source": null,
      "text": "# 0 \"\" 2"
    },
    {
      "labels": [],
      "source": null,
      "text": "#NO_APP"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        mov     eax, DWORD PTR -4[rbp]"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 4
      },
      "text": "        imul    eax, DWORD PTR -4[rbp]"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        pop     rbp"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        ret"
    }
  ],
  "labelDefinitions": {
    "_Z6squarei": 1
  }
}
